terraform: switch to pd-balanced, reduce size to save cost

minrk opened the pull request:
from 1T. Should save ~$1k/month

Ignore the version bumps, which are only to make terraform match what's already been auto-upgraded to, so the new pool has the same version as what's already deployed.

This creates a new user pool to replace the old one, but leaves the existing one with an autoscale max of 1 so it will slowly drain (will need some help with cordoning). Once it's drained, another PR can actually delete the old pool.

I'll do the apply after this is merged. I've synced the versions, but not created the new pool.

xref: [cost calculations](https://github.com/jupyterhub/team-compass/issues/463#issuecomment-998781376)

minrk commented on terraform/prod/main.tf:
```diff
@@ -71,6 +79,50 @@ resource "google_container_node_pool" "user" {
 
+  autoscaling {
+    min_node_count = 0
+    max_node_count = 1
```
This should not trigger immediate scale-down. Autoscale doesn't force these bounds to be satisfied continuously. Instead, it should only prevent scale-up.

yuvipanda commented:
Would this cost us performance in terms of docker pull throughput? That was why we made them big SSDs to begin with, as performance unfortunately scales linearly with size.

The newer `pd-balanced` type is also something to try to cut cost. That's what I now use on my berkeley clusters.

minrk commented:
> Would this cost us performance in terms of docker pull throughput?

I have no idea. I don't think we've done any measuring on how much Disk IO limits pulls vs network IO. [gcloud docs](https://cloud.google.com/compute/docs/disks/performance#cpu_count_size) suggest a 512GB SSD should have 384/768 MBps read/write performance. That seems very fast! But it's ~1/2 what we should have now.

I think we should probably switch to pd-balanced and see what happens. Switching to pd-balanced without losing size would save 40%. We could save the same 50% if we dropped to 800, so I did that. I didn't know about pd-balanced! I switched the core pool to that as well.

minrk commented:
OK, I'll give this a go.
